plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java

```diff
@@ -250,10 +258,14 @@ public CNode describe(T instance, ConfigurationContext context) throws Exception
             Object value = a.getValue(instance);
             if (value != null) {
                 Object converted = Stapler.CONVERT_UTILS.convert(value, a.getType());
-                if (converted instanceof List || !a.isMultiple()) {
+                if (converted instanceof Collection || !a.isMultiple()) {
                     args[i] = converted;
                 } else {
-                    args[i] = Collections.singletonList(converted);
+                    if (Set.class.isAssignableFrom(p.getType())) {
```

Comment:
suggestion reduce nesting by using else if instead of else

Reply:
Done!